src/Sulu/Bundle/AdminBundle/Admin/Routing/FormRouteBuilder.php

```diff
@@ -0,0 +1,126 @@
+        $this->route->addOption('locales', $newLocales);
+
+        return $this;
+    }
+
+    public function setTabTitle(string $tabTitle): FormRouteBuilderInterface
+    {
+        $this->route->addOption('tabTitle', $tabTitle);
+
+        return $this;
+    }
+
+    public function addToolbarActions(array $toolbarActions): FormRouteBuilderInterface
+    {
+        $oldToolbarActions = $this->route->getOption('toolbarActions');
+        $newToolbarActions = $oldToolbarActions ? array_merge($oldToolbarActions, $toolbarActions) : $toolbarActions;
+        $this->route->addOption('toolbarActions', $newToolbarActions);
+
+        return $this;
+    }
+
+    public function addRouterAttributesToFormStore(array $routerAttributesToFormStore): FormRouteBuilderInterface
+    {
+        $oldRouterAttributesToFormStore = $this->route->getOption('routerAttributesToFormStore');
+        $newRouterAttributesToFormStore = $oldRouterAttributesToFormStore ? array_merge($oldRouterAttributesToFormStore, $routerAttributesToFormStore) : $routerAttributesToFormStore;
+        $this->route->addOption('routerAttributesToFormStore', $newRouterAttributesToFormStore);
+
+        return $this;
+    }
+
+    public function addRouterAttributesToEditRoute(array $routerAttributesToEditRoute): FormRouteBuilderInterface
+    {
+        $oldRouterAttributesToEditRoute = $this->route->getOption('routerAttributesToEditRoute');
+        $newRouterAttributesToEditRoute = $oldRouterAttributesToEditRoute ? array_merge($oldRouterAttributesToEditRoute, $routerAttributesToEditRoute) : $routerAttributesToEditRoute;
+        $this->route->addOption('routerAttributesToEditRoute', $newRouterAttributesToEditRoute);
+
+        return $this;
+    }
+
+    public function setEditRoute(string $editRoute): FormRouteBuilderInterface
+    {
+        $this->route->addOption('editRoute', $editRoute);
+
+        return $this;
+    }
+
+    public function setBackRoute(string $backRoute): FormRouteBuilderInterface
+    {
+        $this->route->addOption('backRoute', $backRoute);
+
+        return $this;
+    }
+
+    public function setIdQueryParameter(string $idQueryParameter): FormRouteBuilderInterface
+    {
+        $this->route->addOption('idQueryParameter', $idQueryParameter);
+
+        return $this;
+    }
+
+    public function setPreviewCondition(string $previewCondition): FormRouteBuilderInterface
+    {
+        $this->route->addOption('preview', $previewCondition);
+
+        return $this;
+    }
+
+    public function setParent(string $parent): FormRouteBuilderInterface
+    {
+        $this->route->setParent($parent);
+
+        return $this;
+    }
+
+    public function getRoute(): Route
+    {
+        if (!$this->route->getOption('resourceKey')) {
+            throw new \DomainException(
+                'A route for a Form view needs a "resourceKey" option.'
+                . ' You have likely forgotten to call the "setResourceKey" method.'
+            );
+        }
+
```

Comment:
Would love to check here other stuff as well, e.g. if the locale is set, but it is not that easy, because of the route inheritance we have, which is resolved at a later stage.

In our current setup e.g. the `:locale` placeholder in the path is set on the `ResourceTabs` route, and the `Form` route inherits that part of the path, but the builder doesn't know about it yet. So it is not possible to check it in this place, except we remove that inheritance stuff (for both options and path).

That would mean we have to write the full path in each route, even if the first part of the parent route could be used, and we would have to duplicate some options across routes. However, this duplication could be fixed by using variables.

I am really thinking about this options, since it would make the system less complex and magical, and therefore easier to understand.